### PR TITLE
[oxygen] Oxygen updating utils and related tests for Unicode round two

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Updating various utils and related tests for Unicode goodness

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by rallytime** [comment]
@garethgreenaway lint: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/18156/violations/ and this appears to be failing one related test, `unit.templates.test_jinja.TestCustomExtensions.test_load_yaml` on PY3: https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu16-py3/5618/

**Comment by garethgreenaway** [comment]
@rallytime Should be good to go.  The one in list_differ.py was missing some lines and the one in jinja.py I swapped out some formatting that should have been changed.

File: salt/utils/iam.py [hunk 1]
```diff
@@ -59,5 +59,5 @@ def _convert_key_to_str(key):
     if isinstance(key, six.text_type):
         # the secret key must be bytes and not unicode to work
         #  properly with hmac.new (see http://bugs.python.org/issue5285)
-        return str(key)
+        return six.text_type(key)
```
Comment (damon-atkins) [inline salt/utils/iam.py]: May be this should call salt.utils.to_bytes

File: salt/utils/jid.py [hunk 2]
```diff
@@ -116,7 +114,7 @@ def jid_dir(jid, job_dir=None, hash_type='sha256'):
     Return the jid_dir for the given job id
     '''
     if not isinstance(jid, six.string_types):
-        jid = str(jid)
+        jid = six.text_type(jid)
     if six.PY3:
         jid = jid.encode('utf-8')
```
Comment (damon-atkins) [inline salt/utils/jid.py]: Maybe these few lines should be replaced with   jid = salt.utils.to_bytes(jid)

**Review by rallytime — APPROVED** [review]

**Comment by damon-atkins** [comment]
Says in the code comments it needs to be bytes and not Unicode, yet the change forces it to be Unicode.  Pls update the code or the comments.
```
     if isinstance(key, six.text_type):
         # the secret key must be bytes and not unicode to work
         #  properly with hmac.new (see http://bugs.python.org/issue5285)
-        return str(key)
+        return six.text_type(key)
```

File: salt/utils/jid.py (hunk 2 above)
Comment (terminalmage) [reply salt/utils/jid.py]: hashlib only requires bytes on Python 3, so this is not necessary.

**Pull request merged**